packages/components/pie-radio-group/src/index.ts

```diff
@@ -11,10 +25,133 @@ const componentSelector = 'pie-radio-group';
+
+    /**
+     * Dispatches a custom event to notify each slotted child radio element
+     * when the radio group is disabled.
+     * @private
+     */
+    private _handleDisabled (): void {
+        this._slottedChildren?.forEach((child) => child.dispatchEvent(new CustomEvent(ON_RADIO_GROUP_DISABLED, {
+            bubbles: false, composed: false, detail: { disabled: this.disabled },
+        })));
+    }
+
+    /**
+     * Unselects all radios that are not the selected value.
+     * @param {string} selectedValue - The value of the currently selected radio.
+     * @private
+     */
+    private handleRadioSelection (selectedValue: string): void {
+        this._slottedChildren?.filter((radio) => !radio.disabled)
+        .forEach((radio) => {
+            if (radio.value === selectedValue) {
+                this.value = selectedValue;
+                radio.checked = true;
+            } else {
+                radio.checked = false;
+            }
+        });
+    }
+
+    /**
+     * Handles the radio change event, updates the selected value, and emits a change event for consumers.
+     * @param {Event} event - The change event from a radio element.
+     * @private
+     */
+    private _handleRadioChange (event: Event): void {
+        event.stopPropagation();
+        const target = event.target as HTMLInputElement;
+        this.handleRadioSelection(target.value);
+        const changedEvent = wrapNativeEvent(event);
+        this.dispatchEvent(changedEvent);
+    }
+
+    /**
+     * Updates the `hasLabel` state when content is added to the label slot.
+     * @param {Event} e - The slotchange event.
+     * @private
+     */
+    private handleSlotChange (e: { target: HTMLSlotElement }): void {
+        const childNodes = e.target.assignedNodes({ flatten: true });
+        this.hasLabel = childNodes.length > 0;
+    }
+
+    /**
+     * Renders the label element inside a legend, wrapping the slot content.
+     * @returns {TemplateResult | typeof nothing} The template for the label slot.
+     * @private
+     */
+    private renderWrappedLabel (): TemplateResult | typeof nothing {
+        return this.hasLabel
+            ? html`<legend><slot name='label' @slotchange=${this.handleSlotChange}></slot></legend>`
+            : html`<slot name='label' @slotchange=${this.handleSlotChange}></slot>`;
+    }
+
+    protected updated (_changedProperties: PropertyValues<this>): void {
+        if (_changedProperties.has('disabled')) {
+            this._handleDisabled();
+        }
+
+        if (_changedProperties.has('value')) {
+            this.handleRadioSelection(this.value);
+        }
+    }
+
+    connectedCallback (): void {
+        super.connectedCallback();
+        this.shadowRoot?.addEventListener('change', this._handleRadioChange.bind(this));
+    }
+
+    disconnectedCallback (): void {
+        super.disconnectedCallback();
+        this.shadowRoot?.removeEventListener('change', this._handleRadioChange);
+    }
```

Comment:
I am almost certain this will not remove the event listener because `this._handleRadioChange` and `this._handleRadioChange.bind(this)` are different things. 

It's a little more code, but using an `AbortController` here would ensure the listener is properly removed without having to worry about passing in the exact same reference to the add and remove listener functions.

We have some examples in Modal if you're interested.

Reply:
nice, i updated the checkbox group too for consistency 🙌